test/add/common/add_test.dart

```diff
@@ -50,6 +50,44 @@ void main() {
       await d.appDir({'foo': '1.2.3'}).validate();
     });
 
+    test('does not modify dev_dependencies while normal dependencies',
+        () async {
+      await servePackages((builder) {
+        builder.serve('foo', '1.2.3');
+        builder.serve('foo', '1.2.2');
+      });
+
+      await d.dir(appPath, [
+        YamlDescriptor('pubspec.yaml', '''
+          name: myapp
+          dependencies: 
+
+          dev_dependencies:
+        ''')
+      ]).create();
+
+      await pubAdd(args: ['foo:1.2.3']);
+
+      await d.cacheDir({'foo': '1.2.3'}).validate();
+      await d.appPackagesFile({'foo': '1.2.3'}).validate();
+
+      final finalPubspec = YamlDescriptor('pubspec.yaml', '''
+          name: myapp
+          dependencies: 
+            foo: 1.2.3
+
+          dev_dependencies:
+        ''');
+
+      await d.dir(appPath, [finalPubspec]).validate();
+      final fullPath = p.join(d.sandbox, appPath, 'pubspec.yaml');
+
+      expect(File(fullPath).existsSync(), true);
+
+      final contents = File(fullPath).readAsStringSync();
+      expect(contents, await finalPubspec.read());
```

Comment:
Can we do substring comparison for more robust tests, like `contains('dev_dependencies:')`